### ISSUE TITLE
Pull request #57 broke our build

### ISSUE THREAD

**Issue by Timple** [orocos-toolchain/ocl#61]
commit fd02e738333bb9071c7ecb9bf7085c3031a20c59
**Merge pull request #57 from snrkiwi/refactor-deployers**

Broke our build. This is the log output:

```
$ catkin build --this
----------------------------------------------------------------------------------
Profile:                     default
Extending:          [cached] /home/amigo/ros/kinetic/system/devel:/opt/ros/kinetic
Workspace:                   /home/amigo/ros/kinetic/system
----------------------------------------------------------------------------------
Source Space:       [exists] /home/amigo/ros/kinetic/system/src
Log Space:          [exists] /home/amigo/ros/kinetic/system/logs
Build Space:        [exists] /home/amigo/ros/kinetic/system/build
Devel Space:        [exists] /home/amigo/ros/kinetic/system/devel
Install Space:      [unused] /home/amigo/ros/kinetic/system/install
DESTDIR:            [unused] None
----------------------------------------------------------------------------------
Devel Space Layout:          linked
Install Space Layout:        None
----------------------------------------------------------------------------------
Additional CMake Args:       None
Additional Make Args:        None
Additional catkin Make Args: None
Internal Make Job Server:    True
Cache Job Environments:      False
----------------------------------------------------------------------------------
Whitelisted Packages:        None
Blacklisted Packages:        None
----------------------------------------------------------------------------------


----------------------------------------------------------------------------------
WARNING: Your current environment's CMAKE_PREFIX_PATH is different from the
cached CMAKE_PREFIX_PATH used the last time this workspace was built.

If you want to use a different CMAKE_PREFIX_PATH you should call `catkin clean`
to remove all references to the previous CMAKE_PREFIX_PATH.

Cached CMAKE_PREFIX_PATH:
	/home/amigo/ros/kinetic/system/devel:/opt/ros/kinetic
Current CMAKE_PREFIX_PATH:
	/home/amigo/ros/kinetic/dev/devel:/home/amigo/ros/kinetic/system/devel:/opt/ros/kinetic
----------------------------------------------------------------------------------

[build] Found '88' packages in 0.0 seconds.                                                                         
[build] Package table is up to date.                                                                                
Starting  >>> ocl                                                                                                   
____________________________________________________________________________________________________________________
Errors     << ocl:make /home/amigo/ros/kinetic/system/logs/ocl/build.make.006.log                                   
/home/amigo/ros/kinetic/system/src/ocl/bin/deployer-funcs.cpp: In member function ‘void OCL::TLSFMemoryPool::shutdown()’:
/home/amigo/ros/kinetic/system/src/ocl/bin/deployer-funcs.cpp:413:56: error: ‘get_overhead_size’ was not declared in this scope
         const size_t overhead = get_overhead_size(rtMem);
                                                        ^
/home/amigo/ros/kinetic/system/src/ocl/bin/deployer-funcs.cpp:415:68: error: ‘get_pool_size’ was not declared in this scope
                   << "TLSF bytes allocated=" << get_pool_size(rtMem)
                                                                    ^
/home/amigo/ros/kinetic/system/src/ocl/bin/deployer-funcs.cpp: In function ‘void OCL::deployerDumpTLSF()’:
/home/amigo/ros/kinetic/system/src/ocl/bin/deployer-funcs.cpp:443:31: error: ‘print_all_blocks_mp’ was not declared in this scope
         print_all_blocks_mp(ff);
                               ^
make[2]: *** [bin/CMakeFiles/deployer.dir/deployer-funcs.cpp.o] Error 1
make[1]: *** [bin/CMakeFiles/deployer.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
/home/amigo/ros/kinetic/system/src/ocl/bin/deployer-funcs.cpp: In member function ‘void OCL::TLSFMemoryPool::shutdown()’:
/home/amigo/ros/kinetic/system/src/ocl/bin/deployer-funcs.cpp:413:56: error: ‘get_overhead_size’ was not declared in this scope
         const size_t overhead = get_overhead_size(rtMem);
                                                        ^
/home/amigo/ros/kinetic/system/src/ocl/bin/deployer-funcs.cpp:415:68: error: ‘get_pool_size’ was not declared in this scope
                   << "TLSF bytes allocated=" << get_pool_size(rtMem)
                                                                    ^
/home/amigo/ros/kinetic/system/src/ocl/bin/deployer-funcs.cpp: In function ‘void OCL::deployerDumpTLSF()’:
/home/amigo/ros/kinetic/system/src/ocl/bin/deployer-funcs.cpp:443:31: error: ‘print_all_blocks_mp’ was not declared in this scope
         print_all_blocks_mp(ff);
                               ^
make[2]: *** [bin/CMakeFiles/rttscript.dir/deployer-funcs.cpp.o] Error 1
make[1]: *** [bin/CMakeFiles/rttscript.dir/all] Error 2
make: *** [all] Error 2
cd /home/amigo/ros/kinetic/system/build/ocl; catkin build --get-env ocl | catkin env -si  /usr/bin/make --jobserver-fds=6,7 -j; cd -
....................................................................................................................
Failed     << ocl:make           [ Exited with code 2 ]                                                             
Failed    <<< ocl                [ 1.5 seconds ]                                                                    
[build] Summary: 0 of 1 packages succeeded.                                                                         
[build]   Ignored:   87 packages were skipped or are blacklisted.                                                   
[build]   Warnings:  None.                                                                                          
[build]   Abandoned: None.                                                                                          
[build]   Failed:    1 packages failed.                                                                             
[build] Runtime: 1.9 seconds total.                                                                                 
[~/ros/kinetic/system/src/ocl] ((toolchain-2.9~3))
```

The previous commit still builds fine.

**Comment by meyerj** [comment]
Did you also upgrade RTT to the latest [toolchain-2.9](https://github.com/orocos-toolchain/rtt/tree/toolchain-2.9) revision and do you have the latest version of [toolchain-2.9](https://github.com/orocos-toolchain/ocl/tree/toolchain-2.9) for OCL? PR https://github.com/orocos-toolchain/rtt/pull/214 added some debugging functions to the real-time memory allocator TLSF and commit https://github.com/orocos-toolchain/ocl/commit/a14578452255aa04b5c608f81d34e03571c85700 disables some output that is only available if TLSF has been built with debugging enabled. See also my comment https://github.com/orocos-toolchain/ocl/pull/57#issuecomment-298045164 in the original PR.

I did a first pre-release of 2.9.0 into ROS Kinetic, which will be announced soon (finally) and also published on the Orocos website. The ROS buildfarm is happy with the latest revisions and the tests run successfully locally.

**Comment by Timple** [comment]
I realize 2.9 is not released yet but I thought I'd give a heads up. I was on toolchain-2.9 on rtt but upon closer look it seemed that we forked the rtt repository (to get rid over [annoying errors](https://github.com/tue-robotics/rtt/commit/e99bae7ebd41cb03fe31edfd6f1ddd84d50813cb)) and that our fork was not up to date.

I overlooked the fact that we used a fork, this issue can be closed, the official repo builds just fine.
